### PR TITLE
chore: add script for easy running common developer maven tasks [skip ci]

### PR DESCRIPTION
**Pull request by manolo**

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+#
+# This script displays a menu for selecting actions
+# to run in this project.
+#
+
+## echo is different for bash or regular sh
+e=`echo -e`
+[ "$e" = -e ] && echo='echo' || echo='echo -e'
```
Comment (tulioag) [inline scripts/run.sh:9]: Nit:  `printf` instead of `echo` should behave the same in all shells

File: scripts/run.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+## by default 4 forks are used (set in pom), but it can be changed
+[ -n "$1" ] && fork="-Dfailsafe.forkCount=$1"
+
+## List all modules and ask for one to the user
+askModule() {
+  [ -z "$modules" ] && modules=`cat pom.xml | grep 'module>vaadin.*parent</module' | sed -e 's,.*<module>,,g' | sed -e 's,-flow-parent.*,,g' | sort | cat -n -`
+  $echo "\nList of Modules\n$modules\nType the number of component:  \c"
+  read module
+  case $module in
+   [1-9]*) module=`echo "$modules" | head -$module | tail -1 | awk '{print $2}'`;;
```
Comment (tulioag) [inline scripts/run.sh:20]: Nit: typing any number over 35 chooses vaadin-upload

**Review by tulioag — COMMENTED** [review]
I got an error when trying to run tests.
```
scripts/run.sh 

List of Options
     1	  one compile  - Compile one component (including tests)                     - mvn clean test-comple -pl component...
     2	  one demo     - Run DevServer on the DEMO of one component                  - mvn jetty:run -pl component ...
     3	  one test  IT - Verify Integration-Tests of one component                   - mvn verify -pl component -Dit.Test=...
     4	  one jetty IT - Start Jetty Server on one component IT module               - mvn jetty:run -pl component ...
     5	  one sauce IT - Verify Integration-Tests of one component in SauceLabs      - mvn verify -pl component -Dsauce.user ...
     6	  all compile  - Compile all components (including tests)                    - mvn clean test-compile ... 
     7	  all install  - Install modules in local maven (demos, addons & testbenchs) - mvn install ...
     8	  all test  IT - Verify merged IT's of all component (takes a while)         - mvn verify -pl integration-tests -Dit.Test=...
     9	  all jetty IT - Start Jetty Server on merged IT's module                    - mvn jetty:run -pl integration-tests ...
    10	  all sauce IT - Run Integration-Tests of all component in SauceLabs         - mvn verify -pl integration-tests -Dsauce.user ...
Your option:  3

List of Modules
     1	vaadin-accordion
     2	vaadin-app-layout
     3	vaadin-avatar
     4	vaadin-board
     5	vaadin-button
     6	vaadin-charts
     7	vaadin-checkbox
     8	vaadin-combo-box
     9	vaadin-confirm-dialog
    10	vaadin-context-menu
    11	vaadin-cookie-consent
    12	vaadin-crud
    13	vaadin-custom-field
    14	vaadin-date-picker
    15	vaadin-date-time-picker
    16	vaadin-details
    17	vaadin-dialog
    18	vaadin-form-layout
    19	vaadin-grid
    20	vaadin-grid-pro
    21	vaadin-icons
    22	vaadin-iron-list
    23	vaadin-list-box
    24	vaadin-login
    25	vaadin-menu-bar
    26	vaadin-notification
    27	vaadin-ordered-layout
    28	vaadin-progress-bar
    29	vaadin-radio-button
    30	vaadin-rich-text-editor
    31	vaadin-select
    32	vaadin-split-layout
    33	vaadin-tabs
    34	vaadin-text-field
    35	vaadin-time-picker
    36	vaadin-upload
Type the number of component:  38
Specify classes to test (separated by comma), empty for all: [all]: 
Do you want to run also Unit Tests y/n [n]: 
Start jetty (answer n if jetty is running in background) ? y/n [y]: 

Running:
mvn verify -q -am -B -pl vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests -Dtest=none  -DskipFrontend  

out 29, 2020 3:48:58 PM com.vaadin.pro.licensechecker.ProKeyValidator validateServerResponse
INFO: Validating license for vaadin-testbench 6.4.0
Starting ChromeDriver 86.0.4240.75 (c69c33933bfc72a159aceb4aeca939eb0087416c-refs/branch-heads/4240@{#1149}) on port 21475
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
out 29, 2020 3:49:00 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
[qtp1081819622-13] ERROR com.vaadin.flow.server.frontend.FrontendUtils - Cannot get the 'META-INF/VAADIN/index.html' from the classpath
[qtp1081819622-13] ERROR com.vaadin.flow.server.DefaultErrorHandler - 
java.io.UncheckedIOException: java.io.IOException: Failed to load content of '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html'. It is required to have '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html' file when using client side bootstrapping.
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler$IndexHtmlHolder.<init>(IndexHtmlRequestHandler.java:261)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler$IndexHtmlHolder.<init>(IndexHtmlRequestHandler.java:253)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.lambda$getCachedIndexHtmlDocument$0(IndexHtmlRequestHandler.java:220)
	at com.vaadin.flow.server.VaadinServletContext.getAttribute(VaadinServletContext.java:71)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.getCachedIndexHtmlDocument(IndexHtmlRequestHandler.java:220)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.synchronizedHandleRequest(IndexHtmlRequestHandler.java:70)
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1539)
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:873)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1623)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:214)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1701)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1668)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:502)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:370)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:267)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to load content of '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html'. It is required to have '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html' file when using client side bootstrapping.
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.getIndexHtmlDocument(IndexHtmlRequestHandler.java:242)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.access$100(IndexHtmlRequestHandler.java:60)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler$IndexHtmlHolder.<init>(IndexHtmlRequestHandler.java:258)
	... 39 more
out 29, 2020 3:49:11 PM com.vaadin.testbench.ScreenshotOnFailureRule failed
INFO: Error screenshot written to: /home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/error-screenshots/testUploadEventOrder[ANY_Chrome_](com.vaadin.flow.component.upload.tests.UploadIT).png
[qtp1081819622-16] ERROR com.vaadin.flow.server.frontend.FrontendUtils - Cannot get the 'META-INF/VAADIN/index.html' from the classpath
[qtp1081819622-16] ERROR com.vaadin.flow.server.DefaultErrorHandler - 
java.io.UncheckedIOException: java.io.IOException: Failed to load content of '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html'. It is required to have '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html' file when using client side bootstrapping.
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler$IndexHtmlHolder.<init>(IndexHtmlRequestHandler.java:261)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler$IndexHtmlHolder.<init>(IndexHtmlRequestHandler.java:253)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.lambda$getCachedIndexHtmlDocument$0(IndexHtmlRequestHandler.java:220)
	at com.vaadin.flow.server.VaadinServletContext.getAttribute(VaadinServletContext.java:71)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.getCachedIndexHtmlDocument(IndexHtmlRequestHandler.java:220)
	at com.vaadin.flow.server.communication.IndexHtmlRequestHandler.synchronizedHandleRequest(IndexHtmlRequestHandler.java:70)
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1539)
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:873)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1623)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:214)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1701)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1668)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:502)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:370)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:267)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:132)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to load content of '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html'. It is required to have '/home/tulio/projects/vaadin-flow-components/vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/frontend//index.html' file when using client side bootstrapping.

```

**Comment by ZheSun88** [comment]
the script works with windows based on provided options.

**Comment by manolo** [comment]
@tulioag Good catch, the method to figure out whether compile the frontend was using an incorrect variable.
Also changed to use printf, and restricted the option numbers to fail when out of range

**Review by tulioag — APPROVED** [review]

**Pull request merged**